Dark Mode: NxDropdown - RSC-1018 - RSC-1022 - RSC-1019

entroform opened the pull request:
https://issues.sonatype.org/browse/RSC-1018
https://issues.sonatype.org/browse/RSC-1022
https://issues.sonatype.org/browse/RSC-1019

EmilyBermudez88 commented:
It looks like the border of `nx-dropdown__toggle` is inheriting the colour for the tertiary button. I checked and in dark mode the colours are different so a new variable will be needed for that. (Not something to worry about yet, but when buttons is merged, you may also need to look at the border width as well, which is 2px for tertiary buttons in dark mode)

EmilyBermudez88 commented on lib/src/components/NxDropdownMenu/NxDropdownMenu.scss:
```diff
@@ -62,12 +62,12 @@
   &:hover {
-    background-color: var(--nx-color-interactive-background-hover);
+    background-color: var(--nx-color-dropdown-menu-background-hover);
     color: var(--nx-color-text-dark);
```
In light mode, the text colour gets darker, but in dark mode the text colour of the the button stays the same. Maybe check with Jason to confirm that's correct? Also it's now `--nx-color-text-stark` 😄

entroform replied:
> In light mode, the text colour gets darker, but in dark mode the text colour of the the button stays the same (this is the same for the active state). Maybe check with Jason to confirm that's correct? Also it's now `--nx-color-text-stark` 😄

Oo I might've missed it but, for me, it looks like the text color is constant on the toggle button 🤔 
THanks! Changed to stark here: 9c6b042a322d38a14db3641edde860082a8ed87e

EmilyBermudez88 replied:
> > In light mode, the text colour gets darker, but in dark mode the text colour of the the button stays the same (this is the same for the active state). Maybe check with Jason to confirm that's correct? Also it's now `--nx-color-text-stark` 😄
> 
> Oo I might've missed it but, for me, it looks like the text color is constant on the toggle button 🤔 THanks! Changed to stark here: [9c6b042](https://github.com/sonatype/sonatype-react-shared-components/commit/9c6b042a322d38a14db3641edde860082a8ed87e)

It's referring to the `.nx-dropdown-button` inside of the dropdown menu (not the toggle). So in light mode, the button's text is nx-color-text regularly, but when hovered, the text changes to nx-color-text-stark on hover. But in the dark mode mocks, the buttons in the dropdown do not change in colour when hovered or clicked.

entroform replied:
> > > In light mode, the text colour gets darker, but in dark mode the text colour of the the button stays the same (this is the same for the active state). Maybe check with Jason to confirm that's correct? Also it's now `--nx-color-text-stark` 😄
> > 
> > 
> > Oo I might've missed it but, for me, it looks like the text color is constant on the toggle button 🤔 THanks! Changed to stark here: [9c6b042](https://github.com/sonatype/sonatype-react-shared-components/commit/9c6b042a322d38a14db3641edde860082a8ed87e)
> 
> It's referring to the `.nx-dropdown-button` inside of the dropdown menu (not the toggle). So in light mode, the button's text is nx-color-text regularly, but when hovered, the text changes to nx-color-text-stark on hover. But in the dark mode mocks, the buttons in the dropdown do not change in colour when hovered or clicked.

Ah thanks, fixed here and confirmed with Jason
84a124a8986b6b13c3b07c33ef88260a6c803168

EmilyBermudez88 commented:
Don't forget the NxFilterInput dark mode snapshots as well 😄

EmilyBermudez88 commented:
The "Reset" button that comes with NxDropdown still has the light mode disabled colours applied to it.

entroform commented:
> It looks like the border of `nx-dropdown__toggle` is inheriting the colour for the tertiary button. I checked and in dark mode the colours are different so a new variable will be needed for that. (Not something to worry about yet, but when buttons is merged, you may also need to look at the border width as well, which is 2px for tertiary buttons in dark mode)

Thanks, it turns out design intended for the toggle button to be based on input not button color:
Fixed here: 4a6140794ad8818e708d0e84be77ad69e8a082e6

entroform commented:
> The "Reset" button that comes with NxDropdown still has the light mode disabled colours applied to it.

Fixed here! a588626dcfe53ea62f397bc949a080aeb67bb875

entroform commented:
> Don't forget the NxFilterInput dark mode snapshots as well 😄

Fixed here!
2bd8cb54e1409b517f20cf44dd6e413967f00002

EmilyBermudez88 commented:
> > It looks like the border of `nx-dropdown__toggle` is inheriting the colour for the tertiary button. I checked and in dark mode the colours are different so a new variable will be needed for that. (Not something to worry about yet, but when buttons is merged, you may also need to look at the border width as well, which is 2px for tertiary buttons in dark mode)
> 
> Thanks, it turns out design intended for the toggle button to be based on input not button color: Fixed here: [4a61407](https://github.com/sonatype/sonatype-react-shared-components/commit/4a6140794ad8818e708d0e84be77ad69e8a082e6)

What are your thoughts on changing it from `border-color` to `border: 1px solid (--nx-color-interactive-border)`? It's only because the `nx-dropdown__toggle` also has `nx-btn--tertiary` and the border will change when we merge NxButton, and will need to be overridden anyway.

EmilyBermudez88 commented:
I think there is a mistake in the mocks, but NxFormSelect has the text set to `indigo-70`, unlike dropdown and the PR which has `indigo-80`

avnvbhatta commented on gallery/src/styles/CssVariables/CssVariablesPage.tsx:
```diff
@@ -257,7 +257,10 @@ const CssVariablesPage = () => {
             <ColorDocRow colorVar="--nx-color-form-element-border">
-              The standard color for the borders of form fields (such as text inputs) in their default state.
+              <NxWarningAlert>
+                Deprecated: This variable is now replaced by <NxCode>--nx-code-interactive-border</NxCode>
```
```suggestion
                Deprecated: This variable is now replaced by <NxCode>--nx-color-interactive-border</NxCode>
```

avnvbhatta commented:
For `NxFormSelect`, the invalid caret is `red-60` in Kaleido, but `red-70` here. Also, valid caret is `green-60` in Kaleido, and `green-70` here.

entroform commented:
> > > It looks like the border of `nx-dropdown__toggle` is inheriting the colour for the tertiary button. I checked and in dark mode the colours are different so a new variable will be needed for that. (Not something to worry about yet, but when buttons is merged, you may also need to look at the border width as well, which is 2px for tertiary buttons in dark mode)
> > 
> > 
> > Thanks, it turns out design intended for the toggle button to be based on input not button color: Fixed here: [4a61407](https://github.com/sonatype/sonatype-react-shared-components/commit/4a6140794ad8818e708d0e84be77ad69e8a082e6)
> 
> What are your thoughts on changing it from `border-color` to `border: 1px solid (--nx-color-interactive-border)`? It's only because the `nx-dropdown__toggle` also has `nx-btn--tertiary` and the border will change when we merge NxButton, and will need to be overridden anyway.

Yup! fixed here: 528f1254e174be38c70a9cc66a1d147d5d0b86df

entroform commented:
> For `NxFormSelect`, the invalid caret is `red-60` in Kaleido, but `red-70` here. Also, valid caret is `green-60` in Kaleido, and `green-70` here.

Thanks for catching that fixed: d28f4f3655e97d4a38ba047565431bb6d2d34850

Will update the visual tests later

EmilyBermudez88 commented on lib/src/components/NxDropdown/NxDropdown.scss:
```diff
@@ -32,6 +32,7 @@
 }
 
 .nx-dropdown__toggle {
+  border: 1px solid (--nx-color-interactive-border);
```
```suggestion
  border: 1px solid var(--nx-color-interactive-border);
```

entroform commented:
> I think there is a mistake in the mocks, but NxFormSelect has the text set to `indigo-70`, unlike dropdown and the PR which has `indigo-80`

Thanks for catching this!
Just talked to Jason, we agreed to stick with indigo-80 for now (matches the other text pattern).
If they decide to change to 70 later we'll revisit it in a separate ticket.

EmilyBermudez88 reviewed (approved):
I think it looks good (minus updating snapshots from https://github.com/sonatype/sonatype-react-shared-components/pull/1033#issuecomment-1489118243). I know there are a LOT of changes here, so maybe a +1 from Abhinav as well to see if he catches any outstanding issues

entroform commented:
Snapshots updated here: c8213a18795a285590cc960b352ddbb219aefe70 😸
